src/main/java/htsjdk/samtools/util/SortingCollection.java

```diff
@@ -153,6 +150,7 @@ private SortingCollection(final Class<T> componentType, final SortingCollection.
         this.codec = codec;
         this.comparator = comparator;
         this.maxRecordsInRam = maxRecordsInRam;
+
         this.ramRecords = (T[])Array.newInstance(componentType, maxRecordsInRam);
```

Comment:
It's not clear to me why this code isn't using `ArrayList` here. There is only a small overhead for using it over a bare array, and it wouldn't require the bogus `componentType` argument to `SortingCollection`.

If the ship has sailed on this API, it would at least be better to use a narrower scope for the `SuppressWarnings` annotation. You can do this by adding an intermediate variable, e.g.
```
        @SuppressWarnings("unchecked") 
        T[] ramRecords = (T[]) Array.newInstance(componentType, maxRecordsInRam);
        this.ramRecords = ramRecords
```